pkg/client/client.go

```diff
@@ -61,6 +65,30 @@ func (c *Client) GetBundleThatProvides(ctx context.Context, group, version, kind
 	return parsedBundle, nil
 }
 
+func (c *Client) Close() error {
+	if c.Conn == nil {
+		return nil
+	}
+	return c.Conn.Close()
+}
+
+func (c *Client) HealthCheck(ctx context.Context, reconnectTimeout time.Duration) (bool, error) {
+	res, err := c.Health.Check(ctx, &grpc_health_v1.HealthCheckRequest{Service: "Registry"})
+	if err != nil {
+		if c.Conn.GetState() == connectivity.TransientFailure {
+			ctx, _ := context.WithTimeout(ctx, reconnectTimeout)
```

Comment:
Why do you discard the cancel function instead of calling it with defer?

Reply:
`WaitForStateChange` blocks, so the only time we'd hit the deferred cancel would be after we either successfully changed states or after the timeout defined here. 

I can call cancel here just for consistency though